pyfr/plugins/acoustics.py

```diff
@@ -0,0 +1,2106 @@
+        rhoinf = self.uinf['rho']
+        uinf = np.array(self.uinf['u'])
+        #Compute soln vars
+        #u' perturbed vel
+        u = np.swapaxes(usoln[:, 1 :ndims+1, :], 1, 2) - uinf 
+        p = usoln[:, -1, :] - pinf # p' fluctuation pressure
+        # density, another formulation will just use rho in usoln[:,0,:]
+        rho_tot = rhoinf + p/(cinf*cinf) 
+        # compute normal velocities
+        un  = np.einsum('ij,kij->ki', qnorms, u) # normal flow velocity
+        Uin = sum((uinf*qnorms).T)  # normal Uinfty
+        vn = un + Uin if self.solidsurf else 0.
+        dvn = vn - Uin  # relative surface velocity
+        dUn = un - dvn  # relative normal flow velocity
+        # compute temporal fluxes
+        rdun = rho_tot*dUn
+        umuinf = u - uinf
+        rhouinf = rhoinf*uinf
+        Q = rdun + rhoinf*dvn
+        F = np.einsum('ij,ijk->ijk', rdun, umuinf)
+        F -= np.einsum('i,j...->j...i', rhouinf, dvn)
+        F += np.einsum('ij,jk->ijk', p, qnorms)
+
+        return Q, F
+        
+    def _compute_observer_pfft(self, Q, F):
+        omega = self.omega        
+        kwv = self.kwv
+        mR = self.magR
+        mRs = self.magRs
+        nR = self.nRvec
+        nRs = self.nRsvec
+        # fluxes signal windowing
+        # the convention is not to subtract the mean if not windowing
+        if self.window:
+            Q -= np.mean(Q, 0)
+            F -= np.mean(F, 0)
+            Q = np.einsum('i,ij->ij', self.wwind, Q)
+            F = np.einsum('i,ijk->ijk', self.wwind, F)
+
+        # perform fft of Q, F fluxes
+        Qfft = np.empty((self.nfreq, self.nqpts), dtype=np.complex64)
+        Ffft = np.empty((self.nfreq, self.nqpts, self.ndims), 
+                                                dtype=np.complex64)
+        F = np.swapaxes(F, 0, 2).reshape(-1, self.ntsub) #shape:nt,nqpts,ndim
+        Qfft = realfft(Q.T).T
+        Ffft = realfft(F).reshape(self.ndims, self.nqpts, -1)
+        Ffft = np.swapaxes(Ffft, 0, 2)
+
+        #compute pfft, i=nob, j=nfreq, k=nqpts, p shape i,j,k 
+        kwvR = np.einsum('ik,j->jik', mR, kwv)
+        exp_term0 = np.exp(-1j*kwvR)        # exp(-ikR)
+        exp_term1 = exp_term0/mRs          # exp(-ikR)/R*
+        exp_term2 = exp_term0/(mRs*mRs) # exp(-ikR)/(R* x R*)
+        # p1_term
+        pfreq  = 1j*np.einsum('j,jik,jk->ijk', omega, exp_term1, Qfft)
+        # p2_term0 
+        pfreq += 1j*np.einsum('j,jik,ikm,jkm->ijk',
+                                        kwv, exp_term1, nR, Ffft)   
+        # p2_term1 
+        pfreq += np.einsum('jik,ikm,jkm->ijk', exp_term2, nRs, Ffft) 
+
+        #surface integration
+        #i=nobservers, j=nfreq or ntime, k=nqpts
+        pfft =  np.einsum('ijk,k->ij', pfreq, self.qdA)
+        pfft /= (4.*np.pi)
+        if self.window:
+            pfft *= self.windscale
+        
+        return pfft
+
+
+class PointAcousticSrc(object):
+    def __init__(self, name, srclocs, flowinfo, *srcdata):
```

Comment:
`*srcdata` pointless.

Reply:
Debug analytical src and different sources could be initialized with different inputs.

Reply:
Sure, but why pass a load of junk to the base class which it is going to do nothing with!?!

Reply:
Just initializing common vars for any point source which are its location and the flowinfo. 
Also, this is just for ease of testing and will probably be removed at the end so no point of modifying it right now.